.eslintrc.js

```diff
@@ -0,0 +1,75 @@
+'use strict'
+
+const common = {
+  env: {
+    node: true,
+    browser: true
+  },
+  rules: {
+    'newline-per-chained-call': 0,
+    'prettier/prettier': 'error',
+    'react-hooks/rules-of-hooks': 'error',
+    'react-hooks/exhaustive-deps': 'warn',
+    'sort-imports': 0
+  },
+  globals: {
+    __DEV__: true
+  },
+  settings: {
+    react: {version: 'detect'}
+  }
+}
+
+module.exports = {
+  ...common,
+  root: true,
+  parser: 'babel-eslint',
+  extends: [
+    './packages/eslint-config-sanity/index.js',
+    './packages/eslint-config-sanity/react.js',
+    './packages/eslint-config-sanity/import.js',
+    'prettier',
+    'prettier/react'
+  ],
+  rules: {
+    ...common.rules,
+    'import/no-extraneous-dependencies': 'off',
+    'import/no-unresolved': [
+      'error',
+      {
+        ignore: ['.*:.*']
+      }
+    ],
+    'import/unambiguous': 'off'
+  },
+  settings: {
+    ...common.settings,
+    'import/ignore': ['\\.css$', '.*node_modules.*', '.*:.*'],
+    'import/resolver': 'webpack'
+  },
+  plugins: ['import', 'prettier', 'react', 'react-hooks'],
+
+  // TypeScript files
+  overrides: [
+    {
+      ...common,
+      files: ['**/*.ts', '**/*.tsx'],
+      parser: '@typescript-eslint/parser',
+      extends: [
+        // './packages/eslint-config-sanity/index.js',
```

Comment:
Isn't this saying not to use any of our own rules?

Reply:
That config is left out, because I copied the setup from the current TS config:
https://github.com/sanity-io/sanity/blob/next/.eslintrc-ts.js#L19-L25

Maybe @bjoerge knows the reason for not including it?

Reply:
I just tried enabling `./packages/eslint-config-sanity/index.js` and running `npm run lint`. Some of the rules in there seem to somehow be incompatible with `@typescript-eslint/parser` (anyways, that's by best guess – as it throws a "TypeError: Cannot read property 'body' of null" exception originating in the eslint source code).

Reply:
Fixed in d6298925 by replacing the `no-useless-constructor` rule with `@typescript-eslint/no-useless-constructor`:
https://github.com/sanity-io/sanity/blob/d629892523030786bef879b3e258b2f0305be74f/.eslintrc.js#L24-L27